ui/src/track-layout/track-layout-slice.ts

```diff
@@ -361,6 +362,10 @@ const trackLayoutSlice = createSlice({
                 ...action,
                 payload: options,
             });
+            state.selectedToolPanelTab = updateSelectedToolPanelTab(
+                state.selection,
+                state.selectedToolPanelTab,
+            );
```

Comment:
Valitun assetin päivitys. Vähän ikävää kylvää näitä ympäriinsä, mutta minkäs teet kun selectionia päivitetään miljoonasta paikasta 🤷

Reply:
Tein tuon `selectedToolPanelTab`:in uudelleennimeämisestä joksikin paremmaksi tiketin eilen (GVT-2994). Näihin voinee koskea sitten sen yhteydessä

Reply:
Joo nämä onThisOrThatChanged funkkarit on aina tuntunut vähän happamilta, mutta menee ehkä ohi tästä tiksusta. 

Noin filosofisesti minusta tänne ei pitäisi kertoa että x muuttui vaan käskeä että aseta x (tyhmä store vs fiksu tilakone) ja lisäksi niistä pitäisi eristää se data ja asettaa se jonain järkevänä kokonaisuutena. Tuo siirtää vähän vastuuta kutsujalle, joten voi mennä toisaalla vähän rumemmaksi, mutta siis ajatuksena esim että ei olisi  onPublicationStateChange + onLayoutContextModeChange + onDesignIdChange vaan vain setLayoutContext. Luulisi että noiden kutsupaikoissa olisi suht helposti tiedossa se koko kontekstikin (eiköhän se ole jo bindattuna ko. komponentteihin muutenkin).